src/Lime/Protocol/Client/Channel.ts

```diff
@@ -86,6 +94,35 @@ abstract class Channel implements MessageChannel, CommandChannel, NotificationCh
   }
   abstract onMessage(message: Message): void;
 
+  processCommand(command: Command, timeout = this.commandTimeout): Promise<Command> {
+    const commandPromise = Promise.race([
+      new Promise((resolve) => {
+        this._commandResolves[command.id] = c => {
+          if (!c.status) return
+
+          resolve(c)
+
+          delete this._commandResolves[command.id]
+        }
+      }),
+      new Promise((_, reject) => {
+        setTimeout(() => {
+          if (!this._commandResolves[command.id]) return
+
+          delete this._commandResolves[command.id]
+          command.status = 'failure'
+          command.timeout = true

```

Comment:
Do not change the received command, communicate with the caller just through the exception.